qiita_db/data.py

```diff
@@ -1167,6 +1167,21 @@ def data_type(self, ret_id=False):
         return data_type[0]
 
     @property
+    def samples(self):
+        """Returns samples available in this processed data
+
+        Returns
+        -------
+        set of str
+            Sample_ids available according to the prep template
+        """
+        # Get the samples available for the prep template
+        pid = PreprocessedData(self.preprocessed_data).prep_template
+        conn_handler = SQLConnectionHandler()
+        sql = "SELECT sample_id FROM qiita.prep_{0}".format(pid)
+        return {x[0] for x in conn_handler.execute_fetchall(sql)}
```

Comment:
Yes, the main benefit is not having to transverse the list several
times. I was thinking on something like this:
https://github.com/biocore/qiita/blob/master/qiita_db/parameters.py#L70